### ISSUE TITLE
dependency on sshpass is crashing computer on osx

### ISSUE THREAD

**Issue by n-h** [arunoda/meteor-up-legacy#1]
Possibly related to issue in sshpass itself here: http://sourceforge.net/p/sshpass/bugs/9/

Googling around I've noticed other apps that use sshpass having the same issue like the following:

https://github.com/ansible/ansible/issues/5007

-- System crash report (happens almost every time I use "mup logs")

panic(cpu 2 caller 0xffffff802f00c42f): "negative open count (c, 16, 1)"@/SourceCache/xnu/xnu-2422.1.72/bsd/miscfs/specfs/spec_vnops.c:2110
Backtrace (CPU 2), Frame : Return Address
0xffffff80af2abb40 : 0xffffff802ee22f69 
0xffffff80af2abbc0 : 0xffffff802f00c42f 
0xffffff80af2abc00 : 0xffffff802f011896 
0xffffff80af2abc50 : 0xffffff802effd2ae 
0xffffff80af2abcc0 : 0xffffff802efdb811 
0xffffff80af2abd10 : 0xffffff802efdaf60 
0xffffff80af2abd50 : 0xffffff802efdba9d 
0xffffff80af2abd80 : 0xffffff802effd9d3 
0xffffff80af2abdd0 : 0xffffff802f1cc676 
0xffffff80af2abe50 : 0xffffff802ee45629 
0xffffff80af2abe80 : 0xffffff802ee486d9 
0xffffff80af2abeb0 : 0xffffff802ee4853e 
0xffffff80af2abee0 : 0xffffff802ee20b53 
0xffffff80af2abf10 : 0xffffff802eedc7d3 
0xffffff80af2abf30 : 0xffffff802eef35d2 

BSD process name corresponding to current thread: ssh

Mac OS version:
13A603

Kernel version:
Darwin Kernel Version 13.0.0: Thu Sep 19 22:22:27 PDT 2013; root:xnu-2422.1.72~6/RELEASE_X86_64
Kernel UUID: -----
Kernel slide:    -----
Kernel text base: -----
System model name: MacBookPro8,1

**Comment by arunoda** [comment]
Did you find any workaround?

On Sat, Dec 7, 2013 at 11:09 AM, n-h notifications@github.com wrote:

> Possibly related to issue in sshpass itself here:
> http://sourceforge.net/p/sshpass/bugs/9/
> 
> Googling around I've noticed other apps that use sshpass having the same
> issue like the following:
> 
> ansible/ansible#5007 https://github.com/ansible/ansible/issues/5007
> 
> -- System crash report (happens almost every time I use "mup logs")
> 
> panic(cpu 2 caller 0xffffff802f00c42f): "negative open count (c, 16,
> 1)"@/SourceCache/xnu/xnu-2422.1.72/bsd/miscfs/specfs/spec_vnops.c:2110
> Backtrace (CPU 2), Frame : Return Address
> 0xffffff80af2abb40 : 0xffffff802ee22f69
> 0xffffff80af2abbc0 : 0xffffff802f00c42f
> 0xffffff80af2abc00 : 0xffffff802f011896
> 0xffffff80af2abc50 : 0xffffff802effd2ae
> 0xffffff80af2abcc0 : 0xffffff802efdb811
> 0xffffff80af2abd10 : 0xffffff802efdaf60
> 0xffffff80af2abd50 : 0xffffff802efdba9d
> 0xffffff80af2abd80 : 0xffffff802effd9d3
> 0xffffff80af2abdd0 : 0xffffff802f1cc676
> 0xffffff80af2abe50 : 0xffffff802ee45629
> 0xffffff80af2abe80 : 0xffffff802ee486d9
> 0xffffff80af2abeb0 : 0xffffff802ee4853e
> 0xffffff80af2abee0 : 0xffffff802ee20b53
> 0xffffff80af2abf10 : 0xffffff802eedc7d3
> 0xffffff80af2abf30 : 0xffffff802eef35d2
> 
> BSD process name corresponding to current thread: ssh
> 
> Mac OS version:
> 13A603
> 
> Kernel version:
> Darwin Kernel Version 13.0.0: Thu Sep 19 22:22:27 PDT 2013;
> root:xnu-2422.1.72~6/RELEASE_X86_64
> Kernel UUID: -----
> Kernel slide: -----
> Kernel text base: -----
> System model name: MacBookPro8,1
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/arunoda/meteor-up/issues/1
> .

## 

Arunoda Susiripala

@arunoda http://twitter.com/arunoda
http://gplus.to/arunodahttps://github.com/arunoda
http://www.linkedin.com/in/arunoda

**Comment by arunoda** [comment]
BTW: Is this only happens for "mup logs", does it work for others?

**Comment by n-h** [comment]
Hey @arunoda I didn't find a workaround when using "ssh-pass", but when not using it, and instead relying on "pem" file it works without any problems since it bypasses ssh-pass.  It happens on all remote actions using ssh-pass (from what I saw).

**Comment by arunoda** [comment]
Yeah. ssh-pass is an issue. I will try to migrate into ssh2: 100% node ssh
module. After that this would get fixed.

On Sunday, December 29, 2013, n-h wrote:

> Hey @arunoda https://github.com/arunoda I didn't find a workaround when
> using "ssh-pass", but when not using it, and instead relying on "pem" file
> it works without any problems since it bypasses ssh-pass. It happens on all
> remote actions using ssh-pass (from what I saw).
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/arunoda/meteor-up/issues/1#issuecomment-31308419
> .

## 

Arunoda Susiripala

@arunoda http://twitter.com/arunoda
http://gplus.to/arunodahttps://github.com/arunoda
http://www.linkedin.com/in/arunoda

**Comment by n-h** [comment]
Awesome

**Comment by th48ad** [comment]
Not sure how Arunoda terminates sshpass during its execution but if it's via the SIGINT signal then I have a fix for sshpass.  There seems to be some race condition when sshpass is abruptly killed via ctrl-c or SIGKILL/SIGTERM it causes a kernel panic in MacOS.  Pretty crappy that a userspace application could trigger that in a kernel..  Anyhow I have a patch for sshpass if you're willing to re-compile from source:

https://gist.github.com/th48ad/276f39ac3cb20f464919

**Comment by liuweichu** [comment]
@th48ad ssh keeps crashing my Air and so I googled here.
I tried your patch, now it sleeps 5s after ^C. However the crashes still happened
